lib/workers/repository/process/vulnerabilities.ts

```diff
@@ -91,51 +97,372 @@ export class Vulnerabilities {
     );
     logger.trace(
       { manager, packageFile, queueLength: queue.length },
-      'fetchManagerPackagerFileUpdates starting with concurrency'
+      'fetchManagerPackageFileVulnerabilities starting with concurrency'
     );
 
     config.packageRules?.push(...(await p.all(queue)).flat());
-    logger.trace({ packageFile }, 'fetchManagerPackagerFileUpdates finished');
+    logger.trace(
+      { packageFile },
+      'fetchManagerPackageFileVulnerabilities finished'
+    );
   }
 
   private async fetchDependencyVulnerabilities(
     packageFileConfig: RenovateConfig & PackageFile,
-    packageDependency: PackageDependency
+    dep: PackageDependency
   ): Promise<PackageRule[]> {
-    const ecosystem =
-      Vulnerabilities.managerEcosystemMap[packageFileConfig.manager!];
+    const ecosystem = Vulnerabilities.datasourceEcosystemMap[dep.datasource!];
+    if (!ecosystem) {
+      logger.trace(`Cannot map datasource ${dep.datasource!} to OSV ecosystem`);
+      return [];
+    }
+
+    let packageName = dep.packageName ?? dep.depName!;
+    if (ecosystem === 'PyPI') {
+      // https://peps.python.org/pep-0503/#normalized-names
+      packageName = packageName.toLowerCase().replace(regEx(/[_.-]+/g), '-');
+    }
+
+    const packageRules: PackageRule[] = [];
+    try {
+      const vulnerabilities = await this.osvOffline?.getVulnerabilities(
+        ecosystem,
+        packageName
+      );
+      if (
+        is.nullOrUndefined(vulnerabilities) ||
+        is.emptyArray(vulnerabilities)
+      ) {
+        logger.trace(
+          `No vulnerabilities found in OSV database for ${packageName}`
+        );
+        return [];
+      }
+
+      const depVersion: string =
+        dep.lockedVersion ?? dep.currentVersion ?? dep.currentValue!;
+
+      const versioning = getDefaultVersioning(dep.datasource);
```

Comment:
I tend to think this should be: `const versioning = dep.versioning ?? getDefaultVersioning(dep.datasource);`

Similar to here: https://github.com/renovatebot/renovate/blob/5c5cde9d780626a9bc9d4d5fd33fdd33bb0d5847/lib/workers/repository/process/lookup/index.ts#L58

Would be great to get confirmation before I set this based on a vague idea...